src/main/java/edition/academy/seventh/display/EagerBookPagination.java

```diff
@@ -0,0 +1,97 @@
+package edition.academy.seventh.display;
+
+import edition.academy.seventh.database.model.BookDto;
+import edition.academy.seventh.service.BookService;
+
+import java.util.*;
+
+/**
+ * Eager implementation of {@link Pagination} interface based on {@link BookDto} object.
+ *
+ * <p>The {@code currentPage}, {@code nextPage}, {@code previousPage}, {@code changePaginationSize}
+ * and {@code changeFilter} methods operate on objects located in cache memory.
+ *
+ * <p>Default {@link PaginationSize} is set to <b>twenty</b> records per page.
+ *
+ * @see LazyBookPagination lazy book pagination implemenation.
+ * @author Kamil Rojek
+ */
+class EagerBookPagination implements Pagination<BookDto> {
+  private List<BookDto> books;
+  private EagerBookPaginationFilterHandler filterHandler;
+  private Map<Integer, List<BookDto>> paginationMap;
+  private PaginationSize paginationSize = PaginationSize.TWENTY;
+  private int currentPage;
+
+  EagerBookPagination(BookService bookService) {
+    this.books = bookService.getBooksFromDatabase();
+    this.filterHandler = new EagerBookPaginationFilterHandler(books);
+    initializePaginationMap(paginationSize);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public List<BookDto> currentPage() {
+    return paginationMap.get(currentPage);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public List<BookDto> nextPage() {
+    if (paginationMap.keySet().size() > currentPage) {
+      return paginationMap.get(++currentPage);
+    }
+    return paginationMap.get(currentPage);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public List<BookDto> previousPage() {
+    if (currentPage > 1) {
+      return paginationMap.get(--currentPage);
+    }
+    return paginationMap.get(currentPage);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public List<BookDto> changePaginationSize(PaginationSize size) {
+    initializePaginationMap(size);
+    return paginationMap.get(currentPage);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public List<BookDto> changeFilter(Filter filter) {
+    books = filterHandler.changeFilter((BookFilter) filter);
+    initializePaginationMap(paginationSize);
+    return currentPage();
+  }
+
+  private void initializePaginationMap(PaginationSize paginationSize) {
```

Comment:
Idąc za tym co jest w "Czystym kodzie" to powinno być zaraz pod 1 użyciem

Reply:
Zmienione, dzięki!